src/runtime_src/core/common/device.h

```diff
@@ -307,6 +315,12 @@ class device : public ishim
   // cache xclbin meta data loaded by this process
   uuid m_xclbin_uuid;
   std::map<axlf_section_kind, std::vector<char>> m_axlf_sections;
+
+  // Store memory mapping information <<cuIdx, argIdx>, grpIdx>
+  std::map<std::pair<uint32_t, uint32_t>, uint32_t> m_grp_map;
+
+  // Store memory group information <grpIdx, <startAddr, size>>
+  std::map<uint32_t, std::pair<uint64_t, uint64_t>> m_grp_info;
```

Comment:
Are these needed as data members?    Seems like temporary implementation data used during construction of modified sections.  I think these should be create and returned by proper function called by user of the data.

Reply:
You are correct. This is a temporary data. 
Currently, from shim.cpp layer updating those data member and from device.cpp we are accessing and updating the mem_topology..
If we want to avoid the temporary data member then we need to access the following path:
device.cpp -> ishim.h -> shim.cpp 
Please suggest if we need to do that.

Reply:
Ok. got it.  I suggest you change register_axlf that takes both the axlf and the pGrpInfo (default nullptr), then merge / refactor xrt_core::device::populate_mem_group_info and register_axlf appropriately.  In xclLoadXclBin you call the modified register_axlf with the extra pGrpInfo argument.

Reply:
ok. This make sense.. I will modify this..